### ISSUE TITLE
Siamese Network & Classification (multiple labels)

### ISSUE THREAD

**Issue by Winslow-Xu** [BVLC/caffe#3894]
I understand that Siamese Network is used for metrics learning so that similar pairs of input data have similar features. What I am trying to do is that, I also want the features learned from CNN to be useful in some classification tasks. Thus, I add two more loss functions at the end of one stream in the Siamese Network. [Here](http://xuqiantong.github.io/trying.png) is a sketch of my network. Unfortunately, the training loss of my network will not decrease, i.e. the network is not learning anything, even though each stream of the network works well when I train them separately. 

Can someone tell me where the problem is? Thanks a lot!

**Comment by seanbell** [comment]
Please ask the mailing list for help with using caffe, thanks!

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
